gk/main.c

```diff
@@ -386,33 +434,77 @@ setup_gk_instance(unsigned int lcore_id, struct gk_config *gk_conf)
 	return ret;
 }
 
+static struct flow_entry *
+add_new_flow_from_policy(
+	struct ggu_policy *policy, struct gk_instance *instance,
+	struct gk_lpm *ltbl, uint32_t rss_hash_val)
+{
+	int ret;
+	struct gk_fib *fib;
+	struct flow_entry *fe = NULL;
+
+	fib = look_up_fib(ltbl, &policy->flow);
+	if (fib == NULL || fib->action != GK_FWD_GRANTOR) {
+		/*
+		 * Drop this solicitation to add
+		 * a policy decision.
+		 */
+		RTE_LOG(ERR, GATEKEEPER,
+			"gk: initialize flow entry error in %s\n",
+			__func__);
+
+		return NULL;
+	}
+
+	/* Create a new flow entry. */
+	ret = rte_hash_add_key_with_hash(
+		instance->ip_flow_hash_table,
+ 		&policy->flow, rss_hash_val);
+	if (ret < 0) {
+		RTE_LOG(ERR, HASH,
+			"The GK block failed to add new key to hash table in %s!\n",
+			__func__);
+		return NULL;
+	}
+
+	fe = &instance->ip_flow_entry_table[ret];
+	/*
+	 * It only fills up GK_GRANTED and GK_DECLINED states,
+	 * so, it doesn't need to call initialize_flow_entry().
+	 */
```

Comment:
This comment likely has to be move to the caller of `add_new_flow_from_policy()` since it doesn't make sense here.